src/runtime/test/suggestion-composer-tests.ts

```diff
@@ -9,96 +9,105 @@
  */
 
 import {assert} from './chai-web.js';
+
+import {Arc} from '../arc.js';
+import {SlotComposer} from '../slot-composer.js';
 import {SuggestionComposer} from '../suggestion-composer.js';
-import {FakeSlotComposer} from '../testing/fake-slot-composer.js';
+import {MockSlotComposer} from '../testing/mock-slot-composer.js';
+import {MockSuggestDomConsumer} from '../testing/mock-suggest-dom-consumer.js';
 import {TestHelper} from '../testing/test-helper.js';
 
 class TestSuggestionComposer extends SuggestionComposer {
-  constructor() {
-    super(null, new FakeSlotComposer({containers: {suggestions: {}}}));
-    this.suggestions = [];
-    this.updatesCount = 0;
-    this.updateResolve = null;
-  }
-
-  setSuggestions(suggestions) {
-    this.suggestions = suggestions;
+  get suggestConsumers() {
```

Comment:
maybe better to just have a public getter for `suggestConsumers` in suggestion-composer, to avoid the need to extend the class in test?

Reply:
I guess the larger question should be, is that internal state necessary to be exposed?  I'm guessing the answer is no; since code works without it.

I'd rephrase the question -- is there a way to replace access to suggestConsumers by looking at the external state of the object only and thus achieve black box testing here...